### PR TITLE
network: Fix memory leak PREF64

### PR DESCRIPTION
**Pull request by ssahani**
may fix https://github.com/systemd/systemd/issues/29062

### CODE REVIEW

File: src/network/networkd-json.c [hunk 1]
```diff
@@ -933,20 +943,22 @@ static int pref64_append_json(Link *link, JsonVariant **v) {
...
-                                               JSON_BUILD_PAIR_UNSIGNED("PrefixLength", i->prefix_len),
-                                               JSON_BUILD_PAIR_FINITE_USEC("LifetimeUSec", i->lifetime_usec),
-                                               JSON_BUILD_PAIR_IN6_ADDR_NON_NULL("ConfigProvider", &i->router)));
+                _cleanup_(json_variant_unrefp) JsonVariant *array = NULL;
+
+                r = build_pref64_json(i, &array);
+                if (r < 0)
+                        return r;
+
+                r = json_variant_append_array(&w, array);
```
Comment (yuwata) [inline src/network/networkd-json.c]: Instead, please use `json_variant_append_arrayb()`.

**Review by yuwata — CHANGES_REQUESTED** [review]

File: src/network/networkd-json.c (hunk 1 above)
Comment (ssahani) [reply src/network/networkd-json.c]: OK

File: src/network/networkd-json.c [hunk 2]
```diff
@@ -933,11 +933,14 @@ static int pref64_append_json(Link *link, JsonVariant **v) {
                 return 0;
 
         SET_FOREACH(i, link->ndisc_pref64) {
-                r = json_build(&array, JSON_BUILD_OBJECT(
-                                               JSON_BUILD_PAIR_IN6_ADDR_NON_NULL("Prefix", &i->prefix),
-                                               JSON_BUILD_PAIR_UNSIGNED("PrefixLength", i->prefix_len),
-                                               JSON_BUILD_PAIR_FINITE_USEC("LifetimeUSec", i->lifetime_usec),
-                                               JSON_BUILD_PAIR_IN6_ADDR_NON_NULL("ConfigProvider", &i->router)));
+                _cleanup_(json_variant_unrefp) JsonVariant *e = NULL;
```
Comment (yuwata) [inline src/network/networkd-json.c]: This is not necessary.

Comment (ssahani) [reply src/network/networkd-json.c]: removed . Thanks

**Review by yuwata — APPROVED** [review]
LGTM. Thank you!

**Comment by evverx** [comment]
Looks like I'm late to the party :-)

The memory leak is gone as far as I can see.

(as usual I think it would be great if it was covered by some basis tests. I kind of get why https://github.com/systemd/systemd/pull/28565 and https://github.com/systemd/systemd/pull/28992 were merged with no tests but now that networkd can send, receive and display PREF64s I'm not sure why it isn't covered)

**Pull request merged**